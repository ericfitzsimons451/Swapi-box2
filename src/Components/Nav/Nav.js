import React from 'react'
import './Nav.scss'
import PropTypes from 'prop-types'

const Nav = ({ fetchPeople, fetchPlanets, fetchVehicles }) => {
    return (
        <div className="nav-container">
            <h1 className="title">SwapiBox</h1>
            <button className='fetchPeopleBtn' onClick={fetchPeople}>People</button>
            <button className='fetchPlanetsBtn' onClick={fetchPlanets}>Planets</button>
            <button className='fetchVehiclesBtn' onClick={fetchVehicles}>Vehicles</button>
            <button>Favorites <span>0</span></button>
        </div>
    )
}



Nav.propTypes = {
    fetchPeople: PropTypes.func,
    fetchPlanets: PropTypes.func,
    fetchVehicles: PropTypes.func
}

export default Nav