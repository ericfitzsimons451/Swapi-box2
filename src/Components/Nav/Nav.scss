.nav-container {
    text-align: center;
    min-height: 100%;
}

.nav-container button {
    width: 140px;
    height: 40px;
    margin: 20px;
    border-radius: 10px;
    box-shadow: 1px 1px 1px black;
    background-color: #D3C381;
    font-size: 1.5rem;
    &:hover {
        cursor: pointer;
    }
}
.title {
    text-align: center;
    font-family: 'Star Jedi', arial;
    color: rgb(181, 35, 45);
    text-decoration-line: underline;
    font-size: 4rem;
}