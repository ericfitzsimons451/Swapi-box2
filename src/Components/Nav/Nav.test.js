import React from 'react'
import { shallow, mount } from 'enzyme'
import Nav from '../Nav/Nav'

describe ('Nav', () => {
    // it('should match the snapshot with 3 functions passed down', () => {
    //     // const title = <h1>SwapiBox</h1>
    //     // const button1 = <button>People</button>
    //     // const button2 = <button>Planets</button>
    //     // const button3 = <button>Vehicles</button>
    //     // const button4 = <button>Favorites <span>0</span></button>

    //     const wrapper = shallow(<Nav
    //                                 fetchPeople={jest.fn()} 
    //                                 fetchPlanets={jest.fn()} 
    //                                 fetchVehicles={jest.fn()} />)

    // expect(wrapper).toMatchSnapshot()
    // // expect(wrapper.contains(button1, button2, button3, button4)).toEqual(true)
    // })

   
    it('should call the fetchPeople prop when clicked', () => {
        const mockFetchPeople = jest.fn()
        const wrapper = mount(<Nav
            fetchPeople={mockFetchPeople} />)
        
        wrapper.find('.fetchPeopleBtn').simulate('click')
        expect(mockFetchPeople).toBeCalled()
    })

    it('should call the fetchPlanets prop when clicked', () => {
        const mockFetchPlanets = jest.fn()
        const wrapper = mount(<Nav fetchPlanets={mockFetchPlanets} />)
        wrapper.find('.fetchPlanetsBtn').simulate('click')
        expect(mockFetchPlanets).toBeCalled()
    })

    it('should call the fetchVehicles prop when clicked', () => {
        const mockFetchVehicles = jest.fn()
        const wrapper = mount(<Nav fetchVehicles={mockFetchVehicles} />)
        wrapper.find('.fetchVehiclesBtn').simulate('click')
        expect(mockFetchVehicles).toBeCalled()
    })


})