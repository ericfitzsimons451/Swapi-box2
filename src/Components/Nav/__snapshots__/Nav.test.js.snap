// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with 3 functions passed down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    fetchPeople={[MockFunction]}
    fetchPlanets={[MockFunction]}
    fetchVehicles={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          SwapiBox
        </h1>,
        <button
          onClick={[MockFunction]}
        >
          People
        </button>,
        <button
          onClick={[MockFunction]}
        >
          Planets
        </button>,
        <button
          onClick={[MockFunction]}
        >
          Vehicles
        </button>,
        <button>
          Favorites 
          <span>
            0
          </span>
        </button>,
      ],
      "className": "nav-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SwapiBox",
          "className": "title",
        },
        "ref": null,
        "rendered": "SwapiBox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites ",
            <span>
              0
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Favorites ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "0",
            },
            "ref": null,
            "rendered": "0",
            "type": "span",
          },
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            SwapiBox
          </h1>,
          <button
            onClick={[MockFunction]}
          >
            People
          </button>,
          <button
            onClick={[MockFunction]}
          >
            Planets
          </button>,
          <button
            onClick={[MockFunction]}
          >
            Vehicles
          </button>,
          <button>
            Favorites 
            <span>
              0
            </span>
          </button>,
        ],
        "className": "nav-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SwapiBox",
            "className": "title",
          },
          "ref": null,
          "rendered": "SwapiBox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites ",
              <span>
                0
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Favorites ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "0",
              },
              "ref": null,
              "rendered": "0",
              "type": "span",
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
